appcenter-crashes/Crashes.d.ts

```diff
@@ -0,0 +1,42 @@
+export const enum UserConfirmation {
+  DONT_SEND = 0,
+  SEND = 1,
+  ALWAYS_SEND = 2
+}
+
+export interface ListenerMap {
+  onBeforeSending?: () => void;
+  onSendingSucceeded?: () => void;
+  onSendingFailed?: () => void;
+  getErrorAttachments?: (report) => Promise<[TextAttachment, FileAttachment]>;
+}
+
+interface TextAttachment {
+  text: string;
+  fileName: string;
+}
+
+interface FileAttachment {
+  data: string;
+  fileName: string;
+  contentType: string;
+}
+
+export interface ErrorAttachmentLog {
+  attachmentWithText(text: string, fileName: string): TextAttachment;
+  attachmentWithBinary(
+    data: object,
+    fileName: string,
+    contentType: string
+  ): FileAttachment;
+}
+
+export default interface Crashes {
+  generateTestCrash(): Promise<void>;
+  hasCrashedInLastSession(): Promise<boolean>;
+  lastSessionCrashReport(): Promise<X>;
```

Comment:
what is X?

Reply:
It's an ErrorReport. Not too familiar with how this works behind the scenes. Looks almost like it's a dictionary/map behind the scenes in both android (https://github.com/Microsoft/AppCenter-SDK-React-Native/blob/develop/appcenter-crashes/android/src/main/java/com/microsoft/appcenter/crashes/AppCenterReactNativeCrashesUtils.java) and ios (https://github.com/Microsoft/AppCenter-SDK-React-Native/blob/develop/appcenter-crashes/ios/AppCenterReactNativeCrashes/AppCenterReactNativeCrashesUtils.m).

Let me know if attempt (coming up soon) is correct